cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -835,6 +845,59 @@ const char* replace_utf8_utf8_utf8(gdv_int64 context, const char* text,
                                              out_len);
 }
 
+FORCE_INLINE
+const char* split_part(gdv_int64 context, const char* text, gdv_int32 text_len,
+                       const char* delimiter, gdv_int32 delim_len, gdv_int32 index,
+                       gdv_int32* out_len) {
+  char* ret;
+  if (index < 1) {
+    gdv_fn_context_set_error_msg(context, "Index should be >= 1");
+    return "";
+  }
+
+  if (delim_len == 0 || text_len == 0) {
+    // output will just be text if no delimiter is provided
+    return text;
+  }
+
+  // converting both c style arrays to string for easy processing
+  std::string input = std::string(text);
```

Comment:
Does this work with utf-8 strings and utf-8 delimiters?

Reply:
currently this will work but after refactor to remove string won't. Will need to handle that separately after removing std::string.

Reply:
Have removed usages of std::string and have also added some unit tests for utf-8 strings and utf-8 delimiters.